lib/data_update_scripts/20200917115821_backfill_listing_published_date.rb

```diff
@@ -0,0 +1,13 @@
+module DataUpdateScripts
+  class BackfillListingPublishedDate
+    def run
+      return unless Listing.column_names.include?("originally_published_at")
+
+      Listing.find_each do |listing|
+        if listing.published
```

Comment:
**suggestion (non-blocking):** Instead of using an `if` here let's just not fetch unpublished listings from the DB:

```ruby
Listing.where(published: true).find_each do |listing|
```

The question is, do we need `ActiveRecord` callbacks to run here? You have a separate indexing script that runs after this one anyway, so maybe we can just do everything in a single update statement without instantiating any Ruby objects:

```ruby
Listing.where(published: true).update_all('originally_published_at = created_at')
```

Maybe we should unify the two scripts into one:

```ruby
def run
  return unless Listing.column_names.include?("originally_published_at")

  Listing.where(published: true).update_all('originally_published_at = created_at')
  Listing.find_each(&:index_to_elasticsearch_inline)
end
```

Reply:
If we use `update!` here then that will trigger all of the listings to be reindexed and then we can remove the  `ReIndexListingWithOriginallyPublishedAt` DataUpdateScript below. I think that is the way to go

Reply:
I have used 
```ruby
Listing.where(published: true).find_each do |listing|

```
with `update!`. I did not use `update_all` since I did not know if it will trigger re-indexing and I have removed separate datascript for re-indexing elastic search. Thank you @mstruve and @citizen428